lib/pecorino.rb

```diff
@@ -38,34 +38,28 @@ def self.prune!
   # @param active_record_schema[ActiveRecord::SchemaMigration] the migration through which we will create the tables
   # @return void
   def self.create_tables(active_record_schema)
-    active_record_schema.create_table :pecorino_leaky_buckets, id: :uuid do |t|
-      t.string :key, null: false
-      t.float :level, null: false
-      t.datetime :last_touched_at, null: false
-      t.datetime :may_be_deleted_after, null: false
-    end
-    active_record_schema.add_index :pecorino_leaky_buckets, [:key], unique: true
-    active_record_schema.add_index :pecorino_leaky_buckets, [:may_be_deleted_after]
+    adapter.create_tables(active_record_schema)
+  end
 
-    active_record_schema.create_table :pecorino_blocks, id: :uuid do |t|
-      t.string :key, null: false
-      t.datetime :blocked_until, null: false
-    end
-    active_record_schema.add_index :pecorino_blocks, [:key], unique: true
-    active_record_schema.add_index :pecorino_blocks, [:blocked_until]
+  def self.adapter=(adapter)
+    @adapter = adapter
```

Comment:
**question:** we're passing adapter class name here and assigning it to `@adapter`.

- I'm wondering, if we need to do any sanity checks on this adapter? if we have such adapter class defined? Maybe it should be a subclass of BaseAdapter?
- Would it be better to pass keys for adapter instead of class? e.g. :sqlite, :postgres or :memory (AcitveRecord uses :postgresql, :sqlite3, and :mysql2)

Reply:
* Sanity checks - not really, since that would be type assertions. If a user decides to assign a `ClownAdapter` into Pecorino - it is their deliberate choice. Subclassing could work but is it really valuable given that the base class does not provide any implementation? Also, subclassing does not have much love from me because if you want to wrap the adapter (for instrumentation etc.) you would need your wrapper to _also_ inherit from that base class. We can document this though!
* Key for adapter - not quite. The fact that we are using a model class allows the adapter to use a different connection to a different database. For example, if you have an ActiveRecord subclass which connects to a `transient-data` DB, passing it in the initialiser allows us to use the connection to `transient-data` to call throttles. This, too, warrants documenting - thanks for pointing that out!